source/hp/dof_handler.cc

```diff
@@ -1143,6 +1166,18 @@ namespace internal
         {
           const auto &fe_transfer = dof_handler.active_fe_index_transfer;
 
+          // Set active_fe_indices on persisting cells.
+          for (const auto &pair : fe_transfer->persisting_cells_fe_index)
+            {
+              const auto &cell = pair.first;
+
+              if (cell->is_locally_owned())
```

Comment:
Do you transfer the `fe_transfer` data object between processors? Because, otherwise, how would you end up with a transfer object for a cell that is not actually locally owned?

Reply:
The `active_fe_index` can be accessed on locally owned and ghost cells, but can be set only on locally owned ones. We wanted to use the same logic on all both sequential and `p::s` cases. (We have a different algorithm for the `p::d` case.)

Thus in the `p::s` case, we cannot set the `active_fe_index` on cells we don't own (correct me when I'm wrong). In a future PR #7933, we will collect fe indices on all non-artificial cells, distribute them only on locally owned ones, and communicate the indices afterwards. This way, we only communicate once, since we no longer have to communicate the indices between collecting and distributing them.
 

To avoid costly communication in case we use a `p::s::Tria` with artificial cells (and to fix a bug), we will relax this conditional and replace it with an `if(!cell->is_artificial())` in #7933 .